carp.common/src/commonMain/kotlin/dk/cachet/carp/common/TimeSpan.kt

```diff
@@ -18,6 +18,8 @@ data class TimeSpan(
     companion object
```

Comment:
I'm really not fond of this class.
- It implements stuff that has been very well implemented before (e.g. java.time)
- Timespan is ambiguous. Duration (amount and unit) and Period (ISO cal) is not. It's really a duration.
- A duration should have a seconds part (long) and a nano second component (long). Best precision and temporal span. Then convert in and out using chrono units

Reply:
Down the road this will be replaced with Kotlin's `Duration`. I looked into that prior to extending this class, but it is still experimental: https://github.com/cph-cachet/carp.core-kotlin/issues/55

Reply:
And, premature optimization, _but_, also worth noting that CARP would never support ns precision (we pinned it at microsecond precision), so doubling the size to represent an otherwise fairly common construct feels unnecessary.